### PR TITLE
[BugFix] Fix iceberg parquet dcheck failed

### PR DESCRIPTION
**Pull request by Smith-Cruise**
Fixes [#3013](https://github.com/StarRocks/StarRocksTest/issues/3013)

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Review by trueeyu — DISMISSED** [review]

File: be/src/formats/parquet/group_reader.cpp [hunk 1]
```diff
@@ -365,10 +365,8 @@ void GroupReader::_collect_field_io_range(const ParquetField& field, const TypeD
                                     active, ranges, end_offset);
         }
     } else if (field.type.type == LogicalType::TYPE_MAP) {
-        // ParquetFiled Map -> Map<Struct<key,value>>
-        DCHECK(field.children[0].type.type == TYPE_STRUCT);
         auto index = 0;
-        for (auto& child : field.children[0].children) {
+        for (auto& child : field.children) {
```
Comment (imay) [inline be/src/formats/parquet/group_reader.cpp:369]: Better to use 
```
for (int i = 0; i < field.children.size(); ++i)
```
BTW, the origin code is `field.children[0].children`, but this is `field.children`, which one is right?

Comment (Smith-Cruise) [reply be/src/formats/parquet/group_reader.cpp]: previous we will parse map as struct<map<key, value>>, now we parse map as map<key, value> directly.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by trueeyu — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by imay — APPROVED** [review]

**Review by trueeyu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Pull request merged**